### PR TITLE
Chore: replace aria-label e2e selectors with data-testid

### PR DESCRIPTION
**Pull request by joshhunt**
Part of https://github.com/grafana/grafana/issues/36523

Replaces many instances of using aria-label incorrectly as a selector for tests.

### CODE REVIEW

**Comment by joshhunt** [comment]
I will be surprised if this works first go! 😬

File: packages/grafana-e2e-selectors/src/selectors/components.ts [hunk 1]
```diff
@@ -14,15 +14,15 @@ export const Components = {
     breadcrumb: (title: string) => `data-testid ${title} breadcrumb`,
   },
   TimePicker: {
-    openButton: 'data-testid TimePicker Open Button',
+    openButton: 'data-testid data-testid TimePicker Open Button',
```
Comment (mckn) [inline packages/grafana-e2e-selectors/src/selectors/components.ts:17]:
```suggestion
    openButton: 'data-testid TimePicker Open Button',
```

**Review by mckn — APPROVED** [review]
LGTM! Added one small nit about a duplication. Once we get this into main it would be good to see that the integration tests in the plugin-examples repo still pass.

**Review by matyax — COMMENTED** [review]
Logs changes look good 👍

**Comment by sunker** [comment]
> Updating _all_ the failing tests is just too much for me. I was foolish to think this would work in one big go 😂
> 
> I'll let teams update their own code.

Are you sure @joshhunt? If this is done in one PR, we may be able to get rid of the `data-test-id` prefix from the selector value, and instead prefix all remaining selectors with `aria-label` since this should be considered legacy. Would need to update the [resolver](https://github.com/grafana/grafana/blob/main/e2e/utils/support/types.ts#L68-L70) then ofc.  

But maybe too much work for one PR? Anyway, appreciate the initiative taken here!

**Comment by joshhunt** [comment]
@sunker I had 103 failing tests in 41 files. Yeah I'm pretty sure I'm not going to do that all myself 😂

**Pull request closed without merging**